engine/src/test/java/org/camunda/bpm/engine/test/bpmn/deployment/BpmnDeploymentTest.java

```diff
@@ -177,10 +189,14 @@ public void shouldLogWarningForDuplicateFilteringWithoutPreviousDeploymentName()
         .addDeploymentResources(deployment.getId());
 
     // when
-    testRule.deploy(deploymentBuilder);
+    DeploymentWithDefinitions deployment2 = testRule.deploy(deploymentBuilder);
     
     // then
     assertEquals(1, loggingRule.getFilteredLog(CMD_LOGGER, "Deployment name set to null. Filtering duplicates will not work properly.").size());
+
+    // cleanup
+    repositoryService.deleteDeployment(deployment.getId());
+    repositoryService.deleteDeployment(deployment2.getId());
```

Comment:
Since we use the `testRule.deploy(...)`, we don't need to clean up here if I am not mistaken?

```suggestion
```